cg/meta/workflow/analysis.py

```diff
@@ -495,10 +503,53 @@ def is_case_ready_for_analysis(self, case_id: str) -> bool:
             return False
         return True
 
+    def does_any_file_need_to_be_retrieved(self, case_id: str) -> bool:
```

Comment:
Would be nice with a unit test as there is much logic

Reply:
Added!